backend/benefit/applications/services/ahjo_integration.py

```diff
@@ -0,0 +1,213 @@
+
+PDF_PATH = os.path.join(os.path.dirname(__file__) + "/pdf_templates")
+TEMPLATE_ID_ASSOCIATION = "associations"
+TEMPLATE_ID_COMPANIES_BENEFIT_WITH_DE_MINIMIS_AID = (
+    "companies_benefit_with_de_minimis_aid"
+)
+TEMPLATE_ID_COMPANIES_BENEFIT_WITHOUT_DE_MINIMIS_AID = (
+    "companies_benefit_without_de_minimis_aid"
+)
+TEMPLATE_ID_COMPOSED_ACCEPTED_PUBLIC = "composed_accepted_public"
+TEMPLATE_ID_COMPOSED_ACCEPTED_PRIVATE = "composed_accepted_private"
+TEMPLATE_ID_COMPOSED_DECLINED_PUBLIC = "composed_declined_public"
+TEMPLATE_ID_COMPOSED_DECLINED_PRIVATE = "composed_declined_private"
+
+JINJA_TEMPLATES_COMPOSED = {
+    TEMPLATE_ID_COMPOSED_ACCEPTED_PUBLIC: {
+        "path": "composed_accepted_public.html",
+        "file_name": "Liite 1 Helsinki-lisä 2021 koontiliite yhteisöille "
+        "mallipohja julkinen.pdf",
+    },
+    TEMPLATE_ID_COMPOSED_ACCEPTED_PRIVATE: {
+        "path": "composed_accepted_private.html",
+        "file_name": "Helsinki-lisä 2021 koontiliite yhteisöille mallipohja "
+        "salassa pidettävä.pdf",
+    },
+    TEMPLATE_ID_COMPOSED_DECLINED_PUBLIC: {
+        "path": "composed_declined_public.html",
+        "file_name": "Helsinki-lisä 2021 kielteiset päätökset hakijakohtainen "
+        "mallipohja {start_number}-{end_number} salassa "
+        "pidettävä.pdf",
+    },
+    TEMPLATE_ID_COMPOSED_DECLINED_PRIVATE: {
+        "path": "composed_declined_private.html",
+        "file_name": "Helsinki-lisä 2021 kielteiset päätökset koontiliite "
+        "yrityksille {start_number}-{end_number} mallipohja salassa "
+        "pidettävä.pdf",
+    },
+}
+
+JINJA_TEMPLATES_SINGLE = {
+    TEMPLATE_ID_ASSOCIATION: {
+        "path": "associations.html",
+        "file_name": "Liite 2 hakijakohtainen mallipohja yhteisöt {app_number}.pdf",
+    },
+    TEMPLATE_ID_COMPANIES_BENEFIT_WITH_DE_MINIMIS_AID: {
+        "path": "companies_benefit_with_de_minimis_aid.html",
+        "file_name": "Liite 2 hakijakohtainen mallipohja de minimis "
+        "yritykset {app_number}.pdf",
+    },
+    TEMPLATE_ID_COMPANIES_BENEFIT_WITHOUT_DE_MINIMIS_AID: {
+        "path": "companies_benefit_without_de_minimis_aid.html",
+        "file_name": "Liite 3 hakijakohtainen mallipohja ei de "
+        "minimis yritykset {app_number}.pdf",
+    },
+}
+
+
+def _get_template(path):
+    template_loader = jinja2.FileSystemLoader(searchpath=PDF_PATH)
+    env = jinja2.Environment(loader=template_loader, autoescape=True)
+    return env.get_template(path)
+
+
+def prepare_pdf_files(apps):
+    pdf_files = []
+    # SINGLE FILE PER APP
+    for app in apps:
+        pdf_files.append(generate_single_file(app))
+
+    # COMPOSED FILES
+    pdf_files += generate_composed_files(apps)
+
+    return pdf_files
+
+
+def generate_single_file(app):
+    # Association
+    if (
+        OrganizationType.resolve_organization_type(app.company.company_form)
+        == OrganizationType.ASSOCIATION
+    ):
```

Comment:
What about associations that do have business activities and have de minimis aid granted? Do they use the same template as other associations?

Reply:
OK, I'll verify the issue with the admins, we can address it in a subsequent PR if needed